src/Native/CMakeLists.txt

```diff
@@ -32,7 +32,6 @@ if(WIN32)
     add_compile_options(/FC) # use full pathnames in diagnostics
     add_compile_options(/DEBUG)
     add_compile_options(/GS)
-    add_compile_options(/W1)
```

Comment:
Is the default level 3?

Reply:
Minor nit - but I would prefer this to be explicitly set.

---
In reply to: [276751232](https://github.com/dotnet/machinelearning/pull/3370#discussion_r276751232) [](ancestors = 276751232)

Reply:
Just an FYI that this code was copied from [https://github.com/dotnet/core-setup/blob/97d93381785a25e433fce74d3dd848f1bb07e0d9/src/settings.cmake#L160](https://github.com/dotnet/core-setup/blob/97d93381785a25e433fce74d3dd848f1bb07e0d9/src/settings.cmake#L160), which is what builds the `dotnet.exe` host. I'm a bit surprised that we've gone through almost 3 major versions of .NET Core using `/W1` and no one has complained.

---
In reply to: [276751232](https://github.com/dotnet/machinelearning/pull/3370#discussion_r276751232) [](ancestors = 276751232)

Reply:
No kidding, I was quite surprised when I found this. Perhaps this was not found because if you look for W3 you will find it and assume that it is set.  Fixing thishas exposed some potentially bad situations.  Perhaps the .NET core script needs to be update as well?

---
In reply to: [276763498](https://github.com/dotnet/machinelearning/pull/3370#discussion_r276763498) [](ancestors = 276763498,276751232)

Reply:
I want explicit setting.

---
In reply to: [276796288](https://github.com/dotnet/machinelearning/pull/3370#discussion_r276796288) [](ancestors = 276796288,276763498,276751232)